### PR TITLE
[HLS-2530] Collect spring rabbit payloads

### PR DESCRIPTION
**Pull request by nozik**

### CODE REVIEW

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java [hunk 1]
```diff
@@ -73,4 +74,15 @@ public Long messagePayloadCompressedSize(Message message) {
   public String messageId(Message message, @Nullable Void unused) {
     return message.getMessageProperties().getMessageId();
   }
+
+  @Nullable
+  @Override
+  public String messagePayload(Message message) {
+    byte[] body = message.getBody();
```
Comment (nata7che) [inline instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java:85]: Add check if message is null?

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/groovy/ContextPropagationTest.groovy [hunk 2]
```diff
@@ -94,7 +94,7 @@ class ContextPropagationTest extends AgentInstrumentationSpecification {
             "$SemanticAttributes.MESSAGING_DESTINATION_KIND" "queue"
             "$SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES" Long
             "$SemanticAttributes.MESSAGING_RABBITMQ_ROUTING_KEY" String
-            "messaging.payload" String
+            "messaging.payload" "test"
```
Comment (nata7che) [inline instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/groovy/ContextPropagationTest.groovy]: This is just a suggestion but maybe change this to "test payload" or something we can quickly understand refers to payload

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/HeliosProcessor.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ public class HeliosProcessor implements SpanProcessor {
   public void onStart(Context parentContext, ReadWriteSpan span) {
     Baggage baggage = Baggage.fromContext(parentContext);
     if (baggage.getEntryValue(HELIOS_TEST_TRIGGERED_TRACE) != null) {
-      span.setAttribute(HELIOS_TEST_TRIGGERED_TRACE, HELIOS_TEST_TRIGGERED_TRACE);
+      span.setAttribute(HELIOS_TEST_TRIGGERED_TRACE, "true");
```
Comment (nata7che) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/HeliosProcessor.java:21]: 😱😱😱😱😍😍😍😍

**Review by nata7che — CHANGES_REQUESTED** [review]

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java (hunk 1 above)
Comment (nozik) [reply instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java:85]: done

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/groovy/ContextPropagationTest.groovy (hunk 2 above)
Comment (nozik) [reply instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/groovy/ContextPropagationTest.groovy]: done

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java [hunk 4]
```diff
@@ -78,11 +78,15 @@ public String messageId(Message message, @Nullable Void unused) {
   @Nullable
   @Override
   public String messagePayload(Message message) {
+    if (message == null) {
+      return null;
+    }
+
     byte[] body = message.getBody();
     if (body != null) {
```
Comment (nata7che) [inline instrumentation/spring/spring-rabbit-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rabbit/SpringRabbitMessageAttributesGetter.java]: you should also flip the condition?

**Review by nata7che — APPROVED** [review]

**Pull request merged**